testutils/src/main/scala/za/co/absa/ComparisonJob.scala

```diff
@@ -0,0 +1,65 @@
+package za.co.absa
+
+import org.apache.spark.sql._
+import org.apache.spark.SparkContext
+import org.apache.spark.sql.types.{StructField, StructType}
+import za.co.absa.enceladus.standardization.CmdConfig
+
+object ComparisonJob {
+  def getReader(cmd: CmdConfig)(implicit sparkSession: SparkSession): DataFrameReader = {
+    val dfReader = sparkSession.read.format(cmd.rawFormat)
+    // applying format specific options
+    val  dfReader4= {
+      val dfReader1 = if (cmd.rowTag.isDefined) dfReader.option("rowTag", cmd.rowTag.get) else dfReader
+      val dfReader2 = if (cmd.csvDelimiter.isDefined) dfReader1.option("delimiter", cmd.csvDelimiter.get) else dfReader1
+      val dfReader3 = if (cmd.csvHeader.isDefined) dfReader2.option("header", cmd.csvHeader.get) else dfReader2
+      dfReader3
+    }
+    if (cmd.rawFormat.equalsIgnoreCase("fixed-width")) {
+      val dfReader5 = if (cmd.fixedWidthTrimValues.get) dfReader4.option("trimValues", "true") else dfReader4
+      dfReader5
+    } else dfReader4
+  }
+
+  def removeMetadataFromSchema(schema: StructType): StructType = {
+    StructType(schema.map{ f =>
+      StructField(f.name, f.dataType, f.nullable)
+    })
+  }
+
+  def main(args: Array[String]): Unit = {
+    val cmd = CmdConfig.getCmdLineArguments(args)
+    val enableWholeStage = false
+
+    implicit val sparkSession: SparkSession = SparkSession.builder()
+      .appName("ABSA QA Test Utils")
+      .config("spark.sql.codegen.wholeStage", enableWholeStage) //disable whole stage code gen - the plan is too long
+      .getOrCreate()
+
+    implicit val sc: SparkContext = sparkSession.sparkContext
+
+    import sparkSession.implicits._
+
+    val expected = getReader(cmd).load(cmd.refPath)
+    val actual = getReader(cmd).load(cmd.stdPath)
+    val expectedSchema = removeMetadataFromSchema(expected.schema)
+    val actualSchema = removeMetadataFromSchema(actual.schema)
+
+    if (expectedSchema != actualSchema) {
+      System.err.println("Expected and actual datasets differ in schemas.")
+      sys.exit(101) // Place holder error code until we get info from DevOps
+    }
+
+    val a: Dataset[Row] = expected.except(actual)
+    val b: Dataset[Row] = actual.except(expected)
+
+    if (a.count() != 0 || b.count() != 0){
+      a.toDF().write.format("parquet").save(s"${cmd.outPath}/expected_on_actual")
+      b.toDF().write.format("parquet").save(s"${cmd.outPath}/actual_on_expected")
+      System.err.println(s"Expected and actual datasets differ. Results written to ${cmd.outPath}")
+      sys.exit(102) // Place holder error code until we get info from DevOps
+    } else {
+      System.out.println("Expected and actual datasets are the same. Carry On.")
```

Comment:
Could do with something more informative, like dataset metrics (e.g., record count and column count)
Also, would skip the `Carry on.`

Reply:
Do we need those metrics? We are running the tests, we should know what we are running it on. Maybe add a Dataset name?

Reply:
It would be nice to have Dataset names, but Dataset names are a Menas concept, we are dealing solely with hdfs paths here.

Reply:
I suppose we can do without the additional metrics being reported. I was just imagining something going horribly wrong and comparing two empty datasets, which are sure to come up the same.

Reply:
And `Carry On.` is a perfectly "regexable" string that you would normally never expect to see in normal output and indicates everything went fine. IMO better then `Test XYZ passed` and you having to input XYZ.

Reply:
Isn't `Expected and actual datasets are the same.` just as regexable by itself?
`Carry On.` is a little too casual/cocky for my liking.

Reply:
`Carry On.` is something that could be across the outputs and jobs. If you have a standard way of saying everything finished ok that is unique to your systems, you have something that is perfectly regexable and always can mean the same. Every job output looks for that `XYZ` line that is basically saying our stuff is ok.

`Expected and actual datasets are the same.` is something that would work only for the job of comparison.

The final line doesn't have to be a `Carry on` if it's too cocky for you. But I am advocating for something unified across our outputs.